javalib/src/main/scala/java/net/HttpCookie.scala

```diff
@@ -0,0 +1,374 @@
+        } else if (propName.regionMatches(
+              true,
+              0,
+              MaxAge,
+              0,
+              MaxAge.length()
+            )) {
+          try {
+            cookie.setMaxAge(java.lang.Long.valueOf(propValue))
+          } catch {
+            case _: NumberFormatException =>
+              throw new IllegalArgumentException(
+                "Illegal cookie max-age attribute"
+              )
+          }
+        } else if (propName.regionMatches(true, 0, Path, 0, Path.length())) {
+          cookie.setPath(propValue)
+        } else if (propName.regionMatches(true, 0, Port, 0, Port.length())) {
+          cookie.setPortlist(propValue)
+        } else if (propName.regionMatches(
+              true,
+              0,
+              Secure,
+              0,
+              Secure.length()
+            )) {
+          cookie.setSecure(true)
+        } else if (propName.regionMatches(
+              true,
+              0,
+              HttpOnly,
+              0,
+              HttpOnly.length()
+            )) {
+          cookie.setHttpOnly(true)
+        } else if (propName.regionMatches(
+              true,
+              0,
+              Version,
+              0,
+              Version.length()
+            )) {
+          try {
+            cookie.setVersion(java.lang.Integer.valueOf(propValue))
+          } catch {
+            case _: NumberFormatException =>
+              throw new IllegalArgumentException(
+                "Illegal cookie max-age attribute"
+              )
+          }
+        }
+
+        i += 1
+
+      }
+
+      cookie
+
+    }
+
+  }
+
+  // NAME = attr
+  // attr = token
+  // token = 1*<any CHAR except CTLs or separators>
+  // separators = "(" | ")" | "<" | ">" | "@"
+  //                | "," | ";" | ":" | "\" | <">
+  //                | "/" | "[" | "]" | "?" | "="
+  //                | "{" | "}" | SP | HT
+  // CHAR = <any US-ASCII character (octets 0 - 127)>
+  // CTL = <any US-ASCII control character (octets 0 - 31) and DEL (127)>
+  private def validateName(name: String): String = {
+    if (name.length == 0)
+      throw new IllegalArgumentException("Illegal cookie name")
+
+    var i = 0
+    while (i < name.length()) {
+      val char = name.charAt(i)
+      if (char < 32 || // CTL
+          char == 127 || // CTL
+          char == '(' ||
+          char == ')' ||
+          char == '<' ||
+          char == '>' ||
+          char == '@' ||
+          char == ',' ||
+          char == ';' ||
+          char == ':' ||
+          char == '\\' ||
+          char == '"' ||
+          char == '/' ||
+          char == '[' ||
+          char == ']' ||
+          char == '?' ||
+          char == '=' ||
+          char == '{' ||
+          char == '}' ||
+          char == ' ' ||
+          char == '\t') {
+        throw new IllegalArgumentException("Illegal cookie name")
+      }
+
+      i += 1
+    }
+
+    name
+
+  }
+
+}
+
```

Comment:
There may need to be a javadoc/scaladoc comment here
so that the documentation build picks this up as a supported class.

@ekrich  What is the current process for a newly supported class
                to get picked up by the documentation?

                 I blinked and the process changed.
                 Thanks.